src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -1,74 +1,97 @@
 package frc.robot.subsystems;
 
 import com.revrobotics.CANSparkBase.ControlType;
+import com.revrobotics.CANSparkBase.IdleMode;
 import com.revrobotics.CANSparkLowLevel.MotorType;
 import com.revrobotics.CANSparkMax;
 import com.revrobotics.RelativeEncoder;
 import com.revrobotics.SparkPIDController;
 
+import edu.wpi.first.networktables.DoublePublisher;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.networktables.PubSubOption;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-
+import frc.lib.lib2706.TunableNumber;
+import frc.robot.Config;
 
 public class Shooter extends SubsystemBase {
-    
-    private static Shooter shooter; 
-     public static Shooter getInstance()
-    {
+    private CANSparkMax m_motor;
+    private SparkPIDController m_pidController;
+    private RelativeEncoder m_encoder;
+
+    private TunableNumber kP = new TunableNumber("Shooter/kP", Config.ShooterConstants.kP);
+    private TunableNumber kI = new TunableNumber("Shooter/kI", Config.ShooterConstants.kI);
+    private TunableNumber kD = new TunableNumber("Shooter/kD", Config.ShooterConstants.kD);
+    private TunableNumber kFF = new TunableNumber("Shooter/kFF", Config.ShooterConstants.kFF);
+ 
+    private DoublePublisher velocityPub;
+
+    private static Shooter shooter;
+    public static Shooter getInstance() {
         if (shooter == null)
             shooter = new Shooter();
- 
         return shooter;
     }
 
-    private CANSparkMax m_motor;
-    private SparkPIDController m_pidController;
-    private RelativeEncoder m_encoder;
-
-    private double kP, kD, kFF, kMaxOutput, kMinOutput, maxRPM; 
-    
-        // add Tunable Number later for ajusting Rpm on the flywheels 
-        public Shooter() {  
-            m_motor = new CANSparkMax(32, MotorType.kBrushless);
+    // add Tunable Number later for ajusting Rpm on the flywheels
+    public Shooter() {
+        System.out.println("[Init] Creating Shooter");
+        m_motor = new CANSparkMax(Config.ShooterConstants.FLY_WHEEL_ID, MotorType.kBrushless);
+        m_motor.restoreFactoryDefaults();
 
-            m_motor.restoreFactoryDefaults();
-            m_motor.setSmartCurrentLimit(70);
-            m_pidController = m_motor.getPIDController();
+        m_motor.setCANTimeout(500); //Units in miliseconds
+        m_motor.setIdleMode(IdleMode.kCoast);
+        m_motor.setInverted(false);
 
-            m_encoder = m_motor.getEncoder();
+        m_pidController = m_motor.getPIDController();
+        m_encoder = m_motor.getEncoder();
+        m_encoder.setAverageDepth(2);//check if 1 would work better
```

Comment:
JavaDocs for setAverageDepth has "This value sets the number of samples in the average for velocity readings."

I would prefer to just remove this line because I don't know how it will effect the velocity readings of the NEO encoder. It might lag the velocity readings a bit since it has to do more samples.

Reply:
Done 👍